memory_engine/memory_store.py

```
@@ -157,7 +157,8 @@ def retrieve(
         Raises:
             ValueError: If query_embedding is empty or top_k is invalid
         """
-        if not query_embedding:
+        # Handle numpy arrays - check None or empty explicitly
+        if query_embedding is None or (hasattr(query_embedding, 'size') and query_embedding.size == 0):
             raise ValueError("Query embedding cannot be empty")
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)